cost/google/cud_report/Google_Committed_UseDiscount_Report.pt

```diff
@@ -0,0 +1,134 @@
+name "Google Committed Use Discount (CUD) Report"
+rs_pt_ver 20180301
+type "policy"
+short_description "Report on all Google CUDs that have been purchased. \n See the [README](https://github.com/rightscale/policy_templates/tree/master/cost/google/cud_report) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
+long_description "Version: 1.0"
+category "Cost"
+severity "low"
+
+##################
+# User inputs    #
+##################
+
+parameter "param_email" do
+  type "list"
+  label "Email addresses of the recipients you wish to notify"
+end
+
+parameter "param_google_project" do
+  type "string"
+  label "Google Cloud Project"
+  allowed_pattern /^[0-9a-z:\.-]+$/
+end
+
+parameter "param_cud_status" do
+ label "CUD status"
+ type "string"
+ allowed_values "All","Active","Expired"
+ default "All"
+end
+
+#auth type: Oauth2, using JWT token.
+auth "auth_google", type: "oauth2" do
+  token_url "https://oauth2.googleapis.com/token"
+  grant type: "jwt_bearer" do
+  iss cred("GC_SA_CLIENT_EMAIL")
+  aud "https://oauth2.googleapis.com/token"
+  additional_claims do {
+      "scope" => "https://www.googleapis.com/auth/compute"
+    } end
+  signing_key cred("GC_SA_PRIVATE_KEY")
+  end
+end
+
+pagination "google_pagination" do
+  get_page_marker do
+    body_path "{{nextPageToken}}"
+  end
+  set_page_marker do
+    query "pageToken"
+  end
+end
+
+#https://cloud.google.com/compute/docs/instances/signing-up-committed-use-discounts#viewing_active_commitments
+datasource "ds_committed_use_discount" do
+  request do
+    auth $auth_google
+    host "www.googleapis.com"
+  header "Content-Type", "application/json"
+    path join(["/compute/v1/projects/",$param_google_project,"/aggregated/commitments"])
+    query "accept" , "application/json"
+  end
+
+  result do
+  encoding "json"
+  collect jmes_path(response, "items.*.commitments[0]") do
+  field "cud_id", jmes_path(col_item, 'id')
+  field "name", jmes_path(col_item, 'name')
+  field "region", jmes_path(col_item, 'region')
+  field "status", jmes_path(col_item, 'status')
+  field "plan", jmes_path(col_item, 'plan')
+  field "endTimestamp", jmes_path(col_item, 'endTimestamp')
+  field "startTimestamp", jmes_path(col_item, 'startTimestamp')
+  end
+ end
+end
+
+datasource "ds_committed_use_discount_map" do
+  run_script $js_committed_use_discount_map, $ds_committed_use_discount, $param_cud_status
+end
+
+#Filter the JSON, based on user selection for param 'param_cud_status' and formating Date.
+script "js_committed_use_discount_map", type: "javascript" do
+  parameters "commitments","param_cud_status"
+  result "content"
+  code <<-EOS
+  console.log(commitments)
```

Comment:
When done debugging remove the console.log statements

Reply:
Thanks, @cdwilhelm  I Removed the console.log.